tests/MongoDB.Driver.Core.Tests/Core/Authentication/SaslPrepHelperTests.cs

```diff
@@ -0,0 +1,153 @@
+        // .NET Standard
+#if NET45
+        [Fact]
+        public void SaslPrepQuery_accepts_undefined_codepoint()
+        {     
+            var strWithUnassignedCodepoint = $"abc{char.ConvertFromUtf32(_unAssignedCodePoint.Value)}";
+            
+            SaslPrepHelper.SaslPrepQuery(strWithUnassignedCodepoint).Should().Be(strWithUnassignedCodepoint);
+        }
+
+        [Fact]
+        public void SaslPrepStored_accepts_RandALCat_Characters_in_first_and_last_position()
+        {
+            SaslPrepHelper.SaslPrepStored("\u0627\u0031\u0627").Should().Be("\u0627\u0031\u0627");
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        [InlineData("A B", "A\u00A0B")]
+        [InlineData("A B", "A\u1680B")]
+        public void SaslPrepStored_maps_space_equivalents_to_space(string expected, string input)
+        {
+            SaslPrepHelper.SaslPrepStored(input).Should().Be(expected);
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        [InlineData("IX", "\u2168")] // "IX", Roman numeral nine
+        [InlineData("IV", "\u2163")] // "IV", Roman numeral four
+        public void SaslPrepStored_returns_expected_output_when_passed_nonNormalized_strings(
+            string expected,
+            string nonNormalizedStr)
+        {
+            SaslPrepHelper.SaslPrepStored(nonNormalizedStr).Should().Be(expected);
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        [InlineData("IX", "I\u00ADX")] // "IX", "I-X"
+        [InlineData("IV", "I\u00ADV")] // "IV", "I-V"
+        public void SaslPrepStored_returns_expected_output_when_passed_partially_SaslPrepped_strings(
+            string expected,
+            string partiallyPreppedStr)
+        {
+            SaslPrepHelper.SaslPrepStored(partiallyPreppedStr).Should().Be(expected);
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        [InlineData("IX", "I\u00ADX")]
+        [InlineData("user", "user")]
+        [InlineData("user=", "user=")]
+        [InlineData("USER", "USER")]
+        [InlineData("a", "\u00AA")]
+        [InlineData("IX", "\u2168")]
+        public void SaslPrepStored_returns_expected_output_when_passed_Rfc4013_examples(string expected, string input)
+        {
+            SaslPrepHelper.SaslPrepStored(input).Should().Be(expected);
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        [InlineData("Prohibited character at position 0", "\u0007")]
+        [InlineData("First character is RandALCat, but last character is not", "\u0627\u0031")]
+        public void SaslPrep_throws_argument_exception_when_passed_Rfc4013_examples(string expectedError, string input)
+        {
+            var exception = Record.Exception(()=>SaslPrepHelper.SaslPrepStored(input));
+            exception.Should().BeOfType<ArgumentException>().Subject.Message.Should().Be(expectedError);
```

Comment:
Remember to put a blank line between each of the arrange/act/assert sections.

(in this case there is no arrange section).